R/adapter.r

```diff
@@ -286,12 +292,12 @@ construct_s3data_adapter <- function() {
   write_function <- function(object, opts) {
     common_s3mpi_package_loader()
 
-    obj <- list(data = switch(1 + is.element("data", names(object$output$options)), 
-                              NULL, object$output$options$data), 
-                label = switch(1 + is.element("label", names(object$output$options)), 
+    obj <- list(data = switch(1 + is.element("data", names(object$output$options)),
```

Comment:
I would avoid the `switch` statement in R in almost all situations. It's slower and less readable. Your code is equivalent to:

```r
object <- list(data = if (is.element("data", names(try(silent = TRUE, object$output$options))) object$output$options$data)
```

Make sure you don't reference `object$output$options` without a `try` or you will break this adapter if people use it for anything except a tundraContainer.